rclcpp/include/rclcpp/qos.hpp

```diff
@@ -65,6 +65,11 @@ class RCLCPP_PUBLIC QoS
     const QoSInitialization & qos_initialization,
     const rmw_qos_profile_t & initial_profile = rmw_qos_profile_default);
 
+  /// Constructor which allows you to construct a Rosout QoS by giving the only required settings.
+  explicit
+  QoS(
+    const rmw_qos_profile_t & initial_profile = rmw_qos_profile_default);
```

Comment:
why not using `QoSInitialization` to introduce `RosoutDefaultQoS` instead of adding this constructor?

Reply:
If the `RosoutDefaultQos` you mentioned is introduced, we should overload the Qos's constructor, because the existing constructor of class Qos cannot statisfy the initialization of rosout's qos profile.

As for why `RosoutDefaultQos` is not introduced to maintain consistency with `parameter_event_qos_`, actually as i saied before, that's what i did at first on this [branch](https://github.com/Ada-King/rclcpp/tree/topic-expose-qos-setting).

This design was proposed by @iuhilnehc-ynos, which i also condider it's better. Not only because code is more concise, what more important is the `RosoutDefaultQos` is not universal, only suitable for topic /rosout.

Reply:
okay thanks, maybe this can be discussed with others.

Reply:
This was intentionally not added before.
Use QoSInitialization::from and a default profile instead of adding this

Reply:
Yes, please remove this constructor, if you want a C++ version of `rcl_qos_profile_rosout_default`, add a new class to the end of this file.